src/browser/modules/Editor/ProjectFilesButton.tsx

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2002-2020 "Neo4j,"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useEffect, useState } from 'react'
+import { connect } from 'react-redux'
+import { withBus } from 'react-suber'
+import { Bus } from 'suber'
+
+import { EditorButton } from 'browser-components/buttons'
+import floppyDisk from 'icons/floppy-disk.svg'
+import * as drawer from 'shared/modules/sidebar/sidebarDuck'
+import { EDIT_CONTENT, SET_CONTENT } from 'shared/modules/editor/editorDuck'
+import {
+  ADD_PROJECT_FILE,
+  SELECT_PROJECT_FILE,
+  IProjectFile,
+  SAVE_PROJECT_FILE,
+  PROJECT_FILE_ERROR,
+  EDIT_PROJECT_FILE_START,
+  EDIT_PROJECT_FILE_END,
+  PROJECT_FILES_MOUNTED,
+  PROJECT_FILES_UNMOUNTED
+} from 'browser/modules/Sidebar/project-files.constants'
+import { useMutation } from '@apollo/client'
+
+const IS_RELATE_AVAILABLE = true // @todo: will come from Redux store eventually
+const PROJECT_EDITOR_BUTTON_TITLE = 'Project File'
+const PROJECT_SAVE_EDITOR_BUTTON_TITLE = 'Save Project File'
+const PROJECT_EDIT_EDITOR_BUTTON_TITLE = 'Edit Project File'
+const PROJECT_FILES_DRAWER_NAME = 'project files'
+
+interface ProjectFilesButtonProps {
+  width?: number
+  editorValue: () => string
+  bus: Bus
+  toggleDrawer: () => void
+  isDrawerOpen: boolean
+}
+
+type IActiveRelateFile = Omit<IProjectFile, 'downloadToken'>
```

Comment:
We don't use the I-prefix in the rest of the project and I don't see the use of it in typescript, so I'd prefer the I-s be omitted

Reply:
Ok fair enough. I like the I prefix as it makes it clear but happy to remove and rename

Reply:
The I prefix is for interfaces by convention :)

Reply:
Id like to keep this too. Thoughts @OskarDamkjaer ?

Reply:
same time, I appreciate that the rest of Browser may not use it so you dont want to add it

Reply:
Well it's not a strong convention in typescript. The official docs don't do it https://www.typescriptlang.org/docs/handbook/interfaces.html

Reply:
I prefer not having the I prefix, for consistency with typescript itself and for reasons listed here: https://stackoverflow.com/questions/31876947/confused-about-the-interface-and-class-coding-guidelines-for-typescript/41967120#41967120

Reply:
Besides, IActiveRelateFile is not even an interface? I think the prefix makes sense in C# but not in typescript, at the very least let's not have a mix of both

Reply:
ok, I will remove the I prefix

Reply:
I agree here, if it's a `type` it should not have a prefix

Reply:
Following up on @HerrEmil comment, we should probably change this eslint rule in relate then and update our code for consistency

Reply:
Ill add a card for this